### PR TITLE
Add DisplayChangeAutomationPipelineTrigger

### PR DESCRIPTION
**Pull request by Maplewings**

### CODE REVIEW

File: LenovoLegionToolkit.Lib.Automation/Steps/ScreenDPIAutomationStep.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+                var dpiInfo = item.GetDisplaScaleInfo();
+
+                if (Log.Instance.IsTraceEnabled)
+                    Log.Instance.Trace($"hw screen name: {name}, current dpi: {dpiInfo.current}");
+
+                if (dpiInfo.current != State.DPI)
+                {
+                    item.ToPathDisplaySource().CurrentDPIScale = (DisplayConfigSourceDPIScale)(uint)State.DPI;
+                    Log.Instance.Trace($"set screen dpi: {State.DisplayName}");
+                    MessagingCenter.Publish(new Notification(NotificationType.ScreenDPISet, NotificationDuration.Long, name));
```
Comment (BartoszCichecki) [inline LenovoLegionToolkit.Lib.Automation/Steps/ScreenDPIAutomationStep.cs:52]: Automation steps should not post notifications. What if you have multiple steps and you get spammed?

**Review by BartoszCichecki — CHANGES_REQUESTED** [review]

File: LenovoLegionToolkit.Lib.Automation/Steps/ScreenDPIAutomationStep.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        public Task RunAsync() => ResetHWMonitorDPI();
+
+
+        private async Task ResetHWMonitorDPI()
+        {
+            foreach (var item in Display.GetDisplays())
+            {
+                var name = item.GetTargetDeviceName();
+                //TODO it should be Equals empty (internal screen)
+                if (!name.StartsWith("HW", StringComparison.CurrentCultureIgnoreCase))
```
[the remark this answers is not in the document]
Comment (Maplewings) [reply LenovoLegionToolkit.Lib.Automation/Steps/ScreenDPIAutomationStep.cs:39]: will change it.

File: LenovoLegionToolkit.Lib.Automation/AutomationProcessor.cs [hunk 3]
```diff
@@ -16,6 +16,7 @@ namespace LenovoLegionToolkit.Lib.Automation;
 public class AutomationProcessor
 {
     private readonly AutomationSettings _settings;
+    private readonly DisplayConfigurationListener _displayConfigurationListener;
```
[the remark this answers is not in the document]
Comment (Maplewings) [reply LenovoLegionToolkit.Lib.Automation/AutomationProcessor.cs:19]: At the begining,  I used WM_DEVICECHANGE event to implement this feature, but sometimes the matepad settings will reset to default, so I have to use DisplaySettingsChanged event to implement it.
If it's not necessary, please ignore it.

File: LenovoLegionToolkit.Lib.Automation/Pipeline/Triggers/DisplayChangeAutomationPipelineTrigger.cs [hunk 4]
```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Threading.Tasks;
+using LenovoLegionToolkit.Lib.Automation.Resources;
+using Newtonsoft.Json;
+
+namespace LenovoLegionToolkit.Lib.Automation.Pipeline.Triggers
+{
+    public class DisplayChangeAutomationPipelineTrigger : IAutomationPipelineTrigger, IDisplayChangeAutomationPipelineTrigger
```
[the remark this answers is not in the document]
Comment (Maplewings) [reply LenovoLegionToolkit.Lib.Automation/Pipeline/Triggers/DisplayChangeAutomationPipelineTrigger.cs:8]: At present, I don't how to catch this event, but I'm still researching and will push these trigger later.

File: LenovoLegionToolkit.Lib/Utils/Compatibility.cs [hunk 5]
```diff
@@ -203,4 +204,20 @@ private static bool GetHasPerformanceModeSwitchingBug(string biosVersion)
...
+        if (_settings?.Store?.DeviceInfo != null)
+        {
+            var deviceInfo = _settings.Store.DeviceInfo;
+            vendor = deviceInfo.Vendor ?? vendor;
+            machineType = deviceInfo.MachineType ?? machineType;
+            model = deviceInfo.Model ?? model;
+            serialNumber = deviceInfo.SerialNumber ?? serialNumber;
+        }
+        return (vendor, machineType, model, serialNumber);
+    }
```
[the remark this answers is not in the document]
Comment (Maplewings) [reply LenovoLegionToolkit.Lib/Utils/Compatibility.cs:222]: will remove it.

[the remark this answers is not in the document]
Comment (Maplewings) [reply LenovoLegionToolkit.Lib/Utils/Compatibility.cs:222]: It's my mistake, I don't know the steps of pull request, so submit extra code to this pull request.

File: LenovoLegionToolkit.Lib.Automation/Steps/ScreenDPIAutomationStep.cs (hunk 1 above)
Comment (Maplewings) [reply LenovoLegionToolkit.Lib.Automation/Steps/ScreenDPIAutomationStep.cs:52]: will remove it

File: LenovoLegionToolkit.Lib.Automation/Pipeline/Triggers/DisplayChangeAutomationPipelineTrigger.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (Maplewings) [reply LenovoLegionToolkit.Lib.Automation/Pipeline/Triggers/DisplayChangeAutomationPipelineTrigger.cs:8]: done.

**Review by Maplewings — COMMENTED** [review]
done

**Pull request closed without merging**